tdb/vidrl_upload.py

```diff
@@ -28,116 +29,120 @@ def read_vidrl(path, fstem, assay_type):
+def convert_vidrl_xls_to_tsv(path, fstem, ind, assay_type):
     import xlrd
-    sera_mapping_file = 'source-data/vidrl_serum_mapping.tsv'
-    sera_mapping = parse_tsv_mapping_to_dict(sera_mapping_file)
     exts = ['.xls', '.xlsm', '.xlsx']
     workbook = xlrd.open_workbook(path+fstem + exts[ind])
-    for sheet in workbook.sheets():
-        tmpfile = 'data/tmp/%s.csv'%(fstem)
-        if not os.path.exists(os.path.dirname(tmpfile)):
-            try:
-                os.makedirs(os.path.dirname(tmpfile))
-            except OSError as exc: # Guard against race condition
-                if exc.errno != errno.EEXIST:
-                    raise
-        with open(tmpfile, 'w') as f:
-            writer = csv.writer(f)
-            writer.writerows(sheet.row_values(row) for row in range(ref_sera_row))
-            # Edit row containing serum strains
-            # if '/' in sheet.row_values(10)[4].strip() and '/' in sheet.row_values(12)[2].strip():
-            #     print(sheet.row_values(12)[4])
-            # else:
-            #     raise ValueError
-            # assume there all always 12 reference viruses in a table
-            row_with_ref_sera = sheet.row_values(ref_sera_row)
-            for i in range(titer_col_start,16):
-                row_with_ref_sera[i] = row_with_ref_sera[i].strip('\n')
-                try:
-                    row_with_ref_sera[i] = sera_mapping[row_with_ref_sera[i].lower()]
-                except KeyError:
-                    print("Couldn't find {} in mapping lookup {}".format(row_with_ref_sera[i], sera_mapping_file))
-                    with open ('data/BAD_VIDRL_KEYS.txt', 'a') as f:
-                        f.write(row_with_ref_sera[i])
-            writer.writerow(row_with_ref_sera)
-            writer.writerows(sheet.row_values(row) for row in range(ref_sera_row+1,sheet.nrows))
-        return
-
-def parse_vidrl_matrix_to_tsv(fname, original_path, assay_type):
-    src_id = fname.split('/')[-1]
-    with open(fname) as infile:
-        csv_reader = csv.reader(infile)
-        mat = list(csv_reader)
-    with open('data/tmp/%s.tsv'%(src_id[:-4]), 'w') as outfile:
-        outfile.write("%s\n" % ("\t".join(ELIFE_COLUMNS)))
-        original_path = original_path.split('/')
-        try:
-            original_path.remove('')
-        except:
-            pass
-        print("assay_type: " + assay_type)
-        if assay_type == "hi":
+
+    # Default patterns, VIDRL
+    virus_pattern = r"[A-Z]/[\w\s-]+/.+/\d{4}"
+    virus_passage_pattern = r"(MDCK|SIAT|E\d+|hCK)"
+    serum_id_pattern = r"^[A-Z]\d{4,8}$"
+    serum_passage_pattern = r"(MDCK\d+|SIAT\d+|E\d+)"
+    serum_abbrev_pattern = r"\w+\s{0,1}\w+/\d+.*"
+    crick = False
+
+    for worksheet in workbook.sheets():
+        # autodetecting titer, strain, serum blocks
+        titer_block = find_titer_block(worksheet)
+
+        if len(titer_block["col_start"]) == 0:
+            print("No titer block found.")
```

Comment:
Thinking about routine/automated usage, down the road a bit, and I bet it's going to be real helpful if the error messages are super explicit about which file/workbook caused an error like this to happen -- so maybe switch the `for worksheet` loop to use `enumerate()` so you can get a loop index var, and include that and the file name in this error message?

Also consider adding `file=sys.stderr` to the `print()`.

Reply:
Thanks! Implemented the `enumerate()` section in https://github.com/nextstrain/fauna/pull/157/commits/80336dbf13f41aee84d2f31c7d2616a18c8133fd. 

I'm thinking whether the error handling should go into titer_block.py, so holding off on it for now.